From a17d2d1f690ebe5d005b4589a5fe378b6487c657 Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Mon, 2 Apr 2018 17:41:50 +0900
Subject: [PATCH] g10: Fix filtering by PK->REQ_USAGE.

* g10/getkey.c (get_pubkey_byfprint): Filter by PK->REQ_USAGE.

--

GnuPG-bug-id: 3844
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 g10/getkey.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/g10/getkey.c b/g10/getkey.c
index dabd052e0..0405d1d21 100644
--- a/g10/getkey.c
+++ b/g10/getkey.c
@@ -1810,6 +1810,8 @@ get_pubkey_byfprint (ctrl_t ctrl, PKT_public_key *pk, kbnode_t *r_keyblock,
       ctx.items[0].mode = fprint_len == 16 ? KEYDB_SEARCH_MODE_FPR16
 	: KEYDB_SEARCH_MODE_FPR20;
       memcpy (ctx.items[0].u.fpr, fprint, fprint_len);
+      if (pk)
+        ctx.req_usage = pk->req_usage;
       rc = lookup (ctrl, &ctx, 0, &kb, &found_key);
       if (!rc && pk)
 	pk_from_block (pk, kb, found_key);
-- 
2.17.0

